### PR TITLE
Table: update column header text to sort and display a tooltip

### PR DESCRIPTION
**Pull request by rjavier-trimbler**
## Description

- TAB navigation order for HEADER > LABEL
- TAB navigation order when Sort prop is enable: HEADER >  LABEL > SORT ICON
- Added tooltip for header Label and Sort Icon  when sort prop is enable
- now the header label behaves the same way as you click the sort Icon to sort the column when Sort prop is enable
- E2E test added.

```
**Focus 
Preconditions: Sort enabled**

- Scenario 1: Validate that header texts are tab focusable when sort is enabled
- Scenario 2: Validate that header texts are tab focusable after adding a new column when sort is enabled
- Scenario 3: Validate that header texts are tab focusable after removing a column when sort is enabled

**Preconditions: Sort disabled**

- Scenario 4: Validate that header texts are tab focusable when sort is disabled
- Scenario 5: Validate that header texts are tab focusable after adding a new column when sort is disabled
- Scenario 6: Validate that header texts are tab focusable after removing a column when sort is disabled

**Sorting**
**Preconditions: Sort enabled**

- Scenario 7: Validate that a column can be sorted ascending when clicking on the header text when the column has not been sorted via the header text
- Scenario 8: Validate that a column can be sorted descending when clicking on the header text when the column has been sorted ascending via the header text
- Scenario 9: Validate that a column can be return to unsorted when clicking on the header text when the column has been sorted descending via  the header text
- Scenario 10: Validate that a column can be sorted ascending when clicking on the header text when the column has been returned to unsorted via the sort icon
- Scenario 11: Validate that a column can be sorted descending when clicking on the header text when the column has been sorted ascending via the sort icon
- Scenario 12: Validate that a column can be return to unsorted when clicking on the header text when the column has been sorted descending via the sort icon

**Tooltips:**

**Preconditions: Sort enabled**

- Scenario 13: validate that header text show the "Sort Ascending" tooltip on a column that have not been used for sorting
- Scenario 14: validate that header text show the "Sorted Ascending" tooltip after clicking the header for sorting for the first time
- Scenario 15: validate that header text show the "Sorted Descending" tooltip after clicking the header for sorting for the second time
- Scenario 16: validate that header text show the "Sort Ascending" tooltip after clicking the header for sorting for the third time
- Scenario 17: validate that only the sort icon tooltip appears when the header text is focused via tab and the sort icon is hovered using the mouse pointer
- Scenario 18: validate that only the sort icon tooltip appears when the header text is focused via tab and the sort icon is hovered using the mouse pointer
- Scenario 19: validate that only the header text tooltip appears when the sort icon is focused via tab and the header text is hovered using the mouse pointer

Tab Order

Scenario 20: Validate tab order when Reorder and Sort are enabled
  1.- Header Cell
  2.- Header Text
  3.- Sort Icon
  4.- Cell border for cell resize
  5.- Back to step 1 but pointing to the next cell to the right, except if its the last column 

Scenario 21: Validate tab order when Reorder is enabled and Sort is disabled
  1.- Header Cell
  2.- Header Text
  4.- Cell border for cell resize
  5.- Back to step 1 but pointing to the next cell to the right, except if its the last column 

Scenario 22: Validate tab order when Reorder is disabled and Sort is enabled
1.- Header Text
2.- Sort Icon
4.- Cell border for cell resize
5.- Back to step 1 but pointing to the next cell to the right, except if its the last column 


```

References #2122

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by jmwood330t — APPROVED** [review]

**Comment by jmwood330t** [comment]
Roliver -

Be sure to get at least one engineer to approve this change ... in addition to me. 

Thanks,
Jim

File: stencil-workspace/src/components/modus-table/parts/columnHeader/modus-table-column-header.tsx [hunk 1]
```diff
@@ -82,90 +82,76 @@ export const ModusTableColumnHeader: FunctionalComponent<ModusTableColumnHeaderP
...
-      <ModusTableColumnResizingHandler
-        table={table}
-        header={header}
-        onMouseEnter={() => onMouseEnterResize()}
-        onMouseLeave={() => onMouseLeaveResize()}
-      />
-    ) : null}
-  </div>
-  )
+      tabindex={`${!isColumnResizing && columnReorder ? '0' : ''}`}
```
Comment (jmwood330t) [inline stencil-workspace/src/components/modus-table/parts/columnHeader/modus-table-column-header.tsx:85]: Big code changes here, and in modus-web-components, we've got all the infrastructure in place for automation testing of the Javascript code. What is the plan for testing these substantial changes? Will there be any automation here?

The BDDs in NAV-422 are quite good, and should be a good foundation for system-level testing. But what about unit or integration level with Javascript test automation?

File: stencil-workspace/src/components/modus-table/modus-table.e2e.ts [hunk 2]
```diff
@@ -1065,4 +1069,145 @@ describe('modus-table', () => {
 
     expect(rowSelectionChange).toHaveReceivedEvent();
   });
+
+
+  describe('Header Text Focus', () => {
```
Comment (jmwood330t) [inline stencil-workspace/src/components/modus-table/modus-table.e2e.ts:1074]: These are good tests, Roliver. I think it'd be a good idea to include the other NAV engineers in your pull requests that have tests in them. I will be pushing them to write Javascript unit tests, and your tests provide great examples for them.

I'll ask Yosuan if he thinks this is a good idea ...

Thanks,
Jim

**Review by jmwood330t — APPROVED** [review]

File: stencil-workspace/src/components/modus-table/modus-table.e2e.ts [hunk 3]
```diff
@@ -1010,62 +1008,60 @@ describe('modus-table', () => {
...
 
-  it('Should display a tooltip when header text is focus after the header got sorted for the third time', async () => {
-    page = await newE2EPage();
-    await page.setContent('<modus-table />');
+      expect(tooltipText).not.toBeNull();
+      expect(tooltipText).toBe('Sorted Descending');
+    });
 
-    const component = await page.find('modus-table');
+    it('Should display a tooltip when header text is focus after the header got sorted for the third time', async () => {
```
Comment (jmwood330t) [inline stencil-workspace/src/components/modus-table/modus-table.e2e.ts:1038]: What testing technique or (techniques) have you used for creating these tests?

**Comment by cjwinsor** [comment]
@Roliver-Javier Instead of moving all the logic up from the `modus-table-sort-icon` into the `modus-table-column-header`, can you instead move the header text into the `modus-table-sort-icon`.

My thinking is you take this old logic
```tsx
        <div class={column.getCanSort() && 'can-sort'}>
          <span class={column.getCanSort() && column.getIsSorted() ? 'sorted' : ''}>{column.columnDef.header}</span>
          {column.getCanSort() && (
            <ModusTableColumnSortIcon
              column={column}
              sortIconStyle={sortIconStyle}
              showSortIconOnHover={showSortIconOnHover}
              isColumnResizing={isColumnResizing}
            />
          )}
        </div>
```

And change it to a function called `renderHeaderText()` where
```tsx
  const renderHeaderText = () => {
    if (column.getCanSort()) {
      return (
        <ModusTableColumnSortIcon
          column={column}
          sortIconStyle={sortIconStyle}
          showSortIconOnHover={showSortIconOnHover}
          isColumnResizing={isColumnResizing}
        />
      );
    }
    return column.columnDef.header;
  };
```

Since `ModusTableColumnSortIcon` already takes the `column` object, all the logic can be moved over there.  Ideally we rename to `ModusTableColumnSortLabel` or something and renaming the file making sure its seen as a rename and not a delete + create so we keep history.  

Then you should be able to easily update `ModusTableColumnSortIcon` to add this in the right spot
```tsx
      <modus-tooltip text={sortingStatus}>
        <span
          onClick={() => column.toggleSorting()}
          onKeyDown={(event) => sortOnKeyDown(column, event)}
          class={`header-text ${column.getIsSorted() ? 'sorted' : ''}`}>
          {column.columnDef.header}
        </span>
      </modus-tooltip>
```

I removed your tabIndex as modus actually didn't want the header text to be tabbed into .. just Cell > Sort Icon > Resize Handle

In that example, I also made a 
```tsx
const sortingStatus = showSortingStatus(column, isColumnResizing);
```

so that string can be shared in the multiple places its used now.  Aside from that, I'm not sure what scenario exists where the TH shouldn't have its tabIndex to 0.  At the moment you left it as `tabindex={`${isColumnResizing ? '' : '0'}`}` but I'm not sure how that affects anything to be honest.

Also you have a lot of logic around disabling tooltips, can you explain why that was needed? Is it needed still after my suggestions?

**Comment by rjavier-trimbler** [comment]
Hi @cjwinsor 
I'll push the refactor changes soon, 

--------------------------------------
About the header Text tabIndex:

as I understand we want to tab navigate through the header text so the order would be:

Cell >>>  
     header text (with tooltip and sort handle) >>>  
               sort Icon (with tooltip and sort handle)  >>> 
                     Resize (resize handle)

PD: there's a discussion already to talk about the Cell tooltip.

---------------------------------
About the disable logic:

this happens when you have several modus-tooltip,

as one popper instance has nothing to do with the other, there are times where the events of each instance is not triggering and then 2 tooltips are shown at the same time, the only way we have to manipulate the instance is via the "disabled" prop of modus-tooltip where it kills the instance, so I have to play with this property by accessing the element and set this property to manipulate multiple instances.

**Comment by cjwinsor** [comment]
@Roliver-Javier I ended up making some changes to remove the disable logic.  We can discuss next week so you can show me the scenario it was needed for. In my testing there were still scenarios where multiple tooltips were shown so I think a global solution is called for outside the scope of this ticket.  I also corrected the e2e tests.  The main thing to know is you can't use multiple `>>>` in a single find, it will ignore everything starting from the second `>>>`.  It just so happened that your tests were passing still though, but after some of my changes I noticed some failures and adjusted accordingly.

**Review by cjwinsor — APPROVED** [review]

**Pull request merged**